### PR TITLE
Add the ability to add configs or secrets from files.

### PR DESCRIPTION
**Pull request by brendandburns**
@itowlson @bnookala @testforstephen

### CODE REVIEW

File: src/configMap.ts [hunk 1]
```diff
@@ -63,3 +65,34 @@ export async function deleteKubernetesConfigFile(kubectl: Kubectl, obj: Kubernet
...
+        fileUris.map((uri) => {
+            const filePath = uri.fsPath;
+            const fileName = basename(filePath);
+            // TODO: I really don't like sync calls here...
+            const buff = fs.readFileToBufferSync(filePath);
+            if (obj.resource == 'configmap') {
+                dataHolder.data[fileName] = buff.toString();
+            } else {
+                dataHolder.data[fileName] = buff.toString('base64');
+            }
```
[the remark this answers is not in the document]
Comment (squillace) [reply src/configMap.ts]: Not so much, no.

[the remark this answers is not in the document]
Comment (squillace) [reply src/configMap.ts]: but... am ok with iterating here.... even if we stomp for the moment. But if we merge, I want an issue opened on it ipso pronto so we don't drop the stomping issue.

**Comment by squillace** [comment]
Like this. Love to merge when we can, and stash an issue to work on stomping.

File: src/configMap.ts [hunk 2]
```diff
@@ -63,3 +65,34 @@ export async function deleteKubernetesConfigFile(kubectl: Kubectl, obj: Kubernet
...
+    } as vscode.OpenDialogOptions);
+    if (fileUris) {
+        console.log(fileUris);
+        const currentNS = await currentNamespace(kubectl);
+        const json = await kubectl.invokeAsync(`get ${obj.resource} ${obj.id} --namespace=${currentNS} -o json`);
+        const dataHolder = JSON.parse(json.stdout);
+        fileUris.map((uri) => {
+            const filePath = uri.fsPath;
+            const fileName = basename(filePath);
+            // TODO: I really don't like sync calls here...
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/configMap.ts]: I just don't like sync i/o in javascript, but I agree the effect is likely minimal here.

File: src/configMap.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/configMap.ts]: Added a warning.

File: src/configMap.ts [hunk 3]
```diff
@@ -63,3 +65,34 @@ export async function deleteKubernetesConfigFile(kubectl: Kubectl, obj: Kubernet
...
+            if (obj.resource == 'configmap') {
+                dataHolder.data[fileName] = buff.toString();
+            } else {
+                dataHolder.data[fileName] = buff.toString('base64');
+            }
+        });
+        const out = JSON.stringify(dataHolder);
+        const shellRes = await kubectl.invokeAsync(`replace -f - --namespace=${currentNS}`, out);
+        if (shellRes.code != 0) {
+            vscode.window.showErrorMessage('Failed to delete file: ' + shellRes.stderr);
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/configMap.ts]: done.

File: src/configMap.ts [hunk 4]
```diff
@@ -63,3 +65,34 @@ export async function deleteKubernetesConfigFile(kubectl: Kubectl, obj: Kubernet
...
+        }
+        explorer.refresh();
+        vscode.window.showInformationMessage(`New data added to resource ${obj.id}.`);
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/configMap.ts]: done,

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

**Pull request merged**